test/integration/user_authentication_test.rb

```diff
@@ -0,0 +1,83 @@
+class UserAuthenticationTest < ActionDispatch::IntegrationTest
+  test "visitor can see login link" do
+    visit root_path
+
+    within("#primary-navigation") do
+      assert page.has_content?("Login")
+    end
+  end
+
+  test "visitor can create account and sign in" do
+    visit root_path
+
+    click_link "Create Account"
+
+    fill_in "Username", with: "Jon"
+    fill_in "Password", with: "password"
+    click_button "Create Account"
+
+    assert_equal '/dashboard', current_path
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.